factory/indexerFactory.go

```diff
@@ -72,10 +78,16 @@ func NewIndexer(args *ArgsIndexerFactory) (indexer.Indexer, error) {
 
 func createElasticProcessor(args *ArgsIndexerFactory) (indexer.ElasticProcessor, error) {
 	databaseClient, err := client.NewElasticClient(elasticsearch.Config{
-		Addresses: []string{args.Url},
-		Username:  args.UserName,
-		Password:  args.Password,
-		Logger:    &logging.CustomLogger{},
+		Addresses:     []string{args.Url},
+		Username:      args.UserName,
+		Password:      args.Password,
+		Logger:        &logging.CustomLogger{},
+		RetryOnStatus: []int{http.StatusConflict},
+		RetryBackoff: func(attempt int) time.Duration {
```

Comment:
this is a little bit ugly and very hard to test.
Suggestion here is to create an interface with one or 2 implementations and proper unit tests

Reply:
I don't think it's worth. Elasticsearch libs expect that argument as  `func(attempt int) time.Duration` . I don't think it is necessary that we create a new interface that exposes a `RetryFunction func(attempt int) time.Duration`

Reply:
It is pretty easy to be implemented. Will create a PR for this

Reply:
done